src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1334,10 +1334,10 @@ private boolean confirmClose(BasePanel panel) {
             } catch (Throwable ex) {
                 return false;
             }
-        } else {
```

Comment:
Your idea here was good, but not completely right: 
You actually had to return true for the case that the save was okay.
```java
   if (response.isPresent() && response.get().equals(saveChanges)) {
            // The user wants to save.
            try {
                SaveDatabaseAction saveAction = new SaveDatabaseAction(panel, Globals.prefs);
                if (!saveAction.save()) {
                    // The action was either canceled or unsuccessful.
                    output(Localization.lang("Unable to save library"));
                    return false;
                } else {
                    return true;
                }
            } catch (Throwable ex) {
                return false;
            }
        }
        return false;
```

Reply:
And while you are there, please add a LOGGER.errror statement for the exception

Reply:
I agree, that is a better place to put the return statement. However, wouldn't both work the same way? 
In case the save fails, it returns from the method,  or if an exception occurs it is caught and again returns false.

Reply:
Even easier: you can remove all return false statements (except the one at the bottom of the function)  and only have the return true in the else case.  And I would switch the if/else for saveDatabase to 
```
if(saveAction.save)`
{
return true; 
{
else
{ output....
}
```
 So when no return true is called,  the function will automatically return false per last statement.

Reply:
But we would also have to take care of the case when the `saveAction.save` throws and exception. 
Also, the last statement could return true as well if the user clicked cancel, but that is already taken care of.

Reply:
I'll make the appropriate changes and push the code.